Makefile

```
@@ -1,5 +1,5 @@
-CC = clang
-CXX = clang++
+CC ?= clang
```

```suggestion
ifeq ($(origin CC), default)
  undefine CC
endif
ifeq ($(origin CXX), default)
  undefine CXX
endif
CC ?= clang
```

make [has a couple of default variables](https://www.gnu.org/software/make/manual/html_node/Implicit-Variables.html), for example CXX=g++. ?= won't override them.

You can use a conditional and the origin function to remove such default variables. It's GNU Make only, but so is the entire concept of conditionals, and we already have a bunch of those.

---

That explains it. @rui314, shall I make the requested changes?

---

Sure, please. I believe that code can be a little bit shorter though.

```
ifeq ($(origin CC), default)
  CC = clang
endif

ifeq ($(origin CXX), default)
  CXX = clang++
endif
```